threedi_scenario_downloader/downloader.py

```diff
@@ -211,11 +217,32 @@ def create_raster_task(
     raster, scenario_instance, projection=None, resolution=None, bbox=None, time=None
 ):
     """create Lizard raster task"""
+
     x1 = scenario_instance["origin_x"]
     y1 = scenario_instance["origin_y"]
     x2 = scenario_instance["upper_bound_x"]
     y2 = scenario_instance["upper_bound_y"]
 
+    # Take bbox dimensions if not bigger than the scenario extent.
+    if bbox:
+        # Check for bounding dictionary
+        if isinstance(bbox, dict):
+            xx1 = bbox["west"]
+            yy1 = bbox["south"]
+            xx2 = bbox["east"]
+            yy2 = bbox["north"]
+        else:
+            xx1, yy1, xx2, yy2 = [float(i) for i in bbox.split(",")]
+        if abs(xx2 - xx1) > abs(x2 - x1):
+            raise ValueError("Chosen bbox wider than scenario extend")
+        elif abs(yy2 - yy1) > abs(y2 - y1):
+            raise ValueError("Chosen bbox taller than scenario extend")
+        else:
+            x1 = xx1
+            y1 = yy1
+            x2 = xx2
+            y2 = yy2
```

Comment:
Na een `raise` knal je toch uit de functie, dus met twee lossse "if" en zonder de "else" is het wat eleganter/duidelijker/normaler:

```suggestion
        if abs(xx2 - xx1) > abs(x2 - x1):
            raise ValueError("Chosen bbox wider than scenario extend")
        if abs(yy2 - yy1) > abs(y2 - y1):
            raise ValueError("Chosen bbox taller than scenario extend")
        x1 = xx1
        y1 = yy1
        x2 = xx2
        y2 = yy2
```

Reply:
Extra nog een inhoudelijke opmerking: je controleert of de bbox breder/hoger is dan het scenario, maar je controleert niet of de bbox niet 300km buiten het gebied ligt. Is dat de bedoeling?